automation/workflow/hook.go

```diff
@@ -0,0 +1,24 @@
+package workflow
+
+import "github.com/qri-io/qri/event"
+
+// "encoding/json"
+
+// HookType is the type of hook
+type HookType string
+
+// A Hook determines under what circumstances its `hook.Event()` should be
+// emitted, and what the event payload should be.
+type Hook interface {
+	// json.Marshaler
+	// json.Unmarshaler
+	ID() ID
```

Comment:
do these need an ID at all? If we're only planning on persisting these in a slice within `workflow.Workflow` then we shouldn't need to worry too much about separate storage.

On the other hand, if the plan is to use these in events to connect an event to which workflow fired, that might be a case for having an ID, but even there, I'm not entirely sure.

Reply:
Yea, I was thinking more that they might be needed to disambiguate different hooks/triggers in the same workflow, but that was the only reason. 

Specifically if we allow cases where a Trigger or Hook can be "updated" without the user handing us the full context of the workflow.

I'm thinking about the difference between the frontend vs the cli. In the frontend, we already have the entire workflow. If a user wants to update a specific hook, we have all the workflow information to send over the api to the backend for us to construct a `*Workflow` and use it in a `store.Update` call.

For the CLI, a similar flow would be maybe first running `qri workflow get` to get a json representation of the `Workflow`. The user could save that to a file, make edits, and then use that file to update the store: `qri workflow update --file`.

In either of these cases, we don't need IDs for the triggers or hooks, because we have the full context of the workflow and the other triggers and hooks.

However, if we give users the ability to 'piecemeal' update a single trigger or hook in the workflow, we may need to keep IDs, especially if we can have multiple triggers or hooks for the same type.
`qri workflow update --id workflow_id --trigger-id trigger_id --periodicity R3/P1W`
Or something like that.

I personally favor the `--file` options because it seems easier to me, to both code and work with, but I haven't done a deep dive on `qri workflow create` and `qri workflow update` possible options.

Reply:
> However, if we give users the ability to 'piecemeal' update a single trigger or hook in the workflow, we may need to keep IDs, especially if we can have multiple triggers or hooks for the same type.

I can't see us going this route. once we allow multiple triggers of the same type, I think users will want to have the store save the order they're in, which requires passing the entire workflow around. Based on this, I vote we drop the ID field on triggers and hooks. If we need disambiguation we should use `workflowID` concatenated with slice index on the workflow struct

Reply:
Sounds good